tests/support/sh_py_run_test.bzl

```diff
@@ -18,162 +18,77 @@ without the overhead of a bazel-in-bazel integration test.
 """
 
 load("@rules_shell//shell:sh_test.bzl", "sh_test")
-load("//python:py_binary.bzl", "py_binary")
-load("//python:py_test.bzl", "py_test")
+load("//python/private:py_binary_macro.bzl", "py_binary_macro")  # buildifier: disable=bzl-visibility
+load("//python/private:py_binary_rule.bzl", "create_binary_rule_builder")  # buildifier: disable=bzl-visibility
```

Comment:
We had a ticket to have such an API, no? Is #1647 about it? We could actually expose the `create_binary_rule_builder` with a statement that this API is very early preview and might change in the subsequent releases.

If we don't want to do it now, we can do it at a later time, but getting feedback from people in that ticket could be a good way to stabilise this.

Reply:
Doh, I now read the reference in the description. The PR is good as it is now.

Reply:
Yes, read my mind! I think the two things to do before that are: use builders for all the args (notably attrs) and create directories in `python/api/` (bzl-per-rule) to be the public access points.

Reply:
SGTM